### PR TITLE
pkg/trace/api: Add symdb upload reverse proxy endpoint

### PR DESCRIPTION
**Pull request by ojung**
### What does this PR do?

This PR adds a new reverse proxy endpoint in the trace-agent API for a new feature called SymDB. The endpoint is very similar in structure to the `/debugger` endpoint and re-uses the transport from that reverse proxy. 

### Describe how to test/QA your changes

Ran the agent locally and sent requests:
```
curl -v -X POST \
  -H "dd-api-key: <api-key>" \
  -H "content-type: multipart/form-data" \
  -F 'event=@event.json;type=application/json' \
  -F 'file=@attachment.json;type=application/json' \
  http://localhost:8126/symdb/v1/input
```

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/apm-symdb-endpoint-ca0c373b539df46d.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    APM: Added a new endpoint in the trace-agent api `/symdb/v1/input` that acts as a reverse proxy forwarding requests to a specific event platform intake. The feature using this is currently in development.
```
Comment (ahmed-mez) [inline releasenotes/notes/apm-symdb-endpoint-ca0c373b539df46d.yaml:11]:
```suggestion
    APM: Added a new endpoint in the trace-agent api `/symdb/v1/input` that acts as a reverse proxy forwarding requests to Datadog. The feature using this is currently in development.
```
Customers don't necessarily understand what event platform intake means.

**Review by ahmed-mez — APPROVED** [review]

File: releasenotes/notes/apm-symdb-endpoint-ca0c373b539df46d.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    APM: Added a new endpoint in the trace-agent api `/symdb/v1/input` that acts as a reverse proxy forwarding requests to Datadog. The feature using this is currently in development.
```
Comment (rtrieu) [inline releasenotes/notes/apm-symdb-endpoint-ca0c373b539df46d.yaml:11]:
```suggestion
    APM: Added a new endpoint in the trace-agent API `/symdb/v1/input` that acts as a reverse proxy forwarding requests to Datadog. The feature using this is currently in development.
```

**Review by rtrieu — APPROVED** [review]
Minor suggestion but approving.

File: pkg/trace/api/symdb.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		Director:  getSymDBDirector(tags, cidProvider, conf.ContainerTags),
+		ErrorLog:  stdlog.New(logger, "symdb.Proxy: ", 0),
+		Transport: transport,
+	}
+}
+
+func getSymDBDirector(tags string, cidProvider IDProvider, containerTags func(string) ([]string, error)) func(*http.Request) {
+	return func(req *http.Request) {
+		req.Header.Set("DD-REQUEST-ID", uuid.New().String())
+		req.Header.Set("DD-EVP-ORIGIN", "agent-symdb")
```
Comment (cimi) [inline pkg/trace/api/symdb.go:77]: What are the requirements for this value, do we need to register it anywhere?

**Review by cimi — APPROVED** [review]
:+1: Thanks!

File: pkg/trace/api/symdb.go (hunk 3 above)
Comment (ojung) [reply pkg/trace/api/symdb.go:77]: It's part of the v2 intake and it's used for observability / debugging purposes. There is no need to register it.

File: cmd/trace-agent/config/config.go [hunk 4]
```diff
@@ -420,6 +420,15 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 	if k := "apm_config.debugger_additional_endpoints"; coreconfig.Datadog.IsSet(k) {
 		c.DebuggerProxy.AdditionalEndpoints = coreconfig.Datadog.GetStringMapStringSlice(k)
 	}
+	if k := "apm_config.symdb_dd_url"; coreconfig.Datadog.IsSet(k) {
+		c.SymDBProxy.DDURL = coreconfig.Datadog.GetString(k)
+	}
+	if k := "apm_config.symdb_api_key"; coreconfig.Datadog.IsSet(k) {
+		c.SymDBProxy.APIKey = coreconfig.Datadog.GetString(k)
+	}
```
[the remark this answers is not in the document]
Comment (ojung) [reply cmd/trace-agent/config/config.go:428]: Agreed and I'm more on the side of being more verbose instead of doing magic behind the scenes.

**Pull request merged**